### ISSUE TITLE
Option to do connection via Proxy

### ISSUE THREAD

**Issue by LeMoussel** [bzikarsky/gelf-php#61]
Would be nice to have an option to initialize HttpTransport / socket-client via Proxy.
Maybe like this :

```
$opts = array('http' => array('proxy' => 'tcp://127.0.0.1:8080', 'request_fulluri' => true));
$context = stream_context_create($opts);
```

**Comment by bzikarsky** [comment]
I think this is a good idea, I propose a solution similar to the one chosen for basic authentication:

- setter/getter for proxy information
- extend private `getContext` to include the information

I'll look into this, but I can't promise a sokution in the next days/weeks. But I'm happy to review/merge a PR with such a feature

**Comment by bzikarsky** [comment]
Implemented, will be part of the 1.6.0 release.
